### PR TITLE
Ting 4259 update dependencies

### PR DESCRIPTION
**Pull request by jvercide-tc**
- update vulnerable dependencies
- add github test CI as a replacement for travic CI
- switch test suite from testee to karma

### CODE REVIEW

File: config/karma.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const karma = require('karma');
+const getPortSync = require('get-port-sync');
+
+const port = getPortSync();
+
+module.exports = function(config) {
+	config.set({
+		autoWatch: false,	// default: true
+		basePath: '../',
+		browserDisconnectTimeout: 45000,	// default: 2000
```
Comment (cjardine-tc) [inline config/karma.js:10]: @jvercide-tc these could likely be reduced significantly as this isn't a large application

File: karma.bootstrap.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+/* global*/
+
+console.info(`********* Karma Boostrap accessed`);	// eslint-disable-line no-console
```
Comment (cjardine-tc) [inline karma.bootstrap.js:4]: @jvercide-tc we may want to add more documentation as to how this works for other using this

File: config/karma.js (hunk 1 above)
Comment (jvercide-tc) [reply config/karma.js:10]: @cjardine-tc updated

File: karma.bootstrap.js (hunk 2 above)
Comment (jvercide-tc) [reply karma.bootstrap.js]: @cjardine-tc added some comments and ref link. also added some FAQs to wiki (https://github.com/tucows/donejs-carousel-plugin/wiki)

File: .gitignore [hunk 3]
```diff
@@ -29,3 +29,9 @@ node_modules
 
 docs/
 dist/
+
+# Pesky DS_Store
+*.DS_Store
+
+# Karma results
+test_results/
```
Comment (cjardine-tc) [inline .gitignore:37]: @jvercide-tc nice, good change, missed that :D

**Comment by jvercide-tc** [comment]
> @jvercide-tc didn't see the changes to the timeout durations but this is still looking good otherwise

@cjardine-tc strange, let me add it again

**Pull request merged**